lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -179,93 +176,20 @@ class TwilioConnectionTransport extends StateMachine {
       }
     });
 
-
-    setupTransport(this);
-
-  }
-
-  /**
-   * Create a Connect, Sync or Disconnect RSP message.
-   * @private
-   * @returns {?object}
-   */
-  _createConnectOrSyncOrDisconnectMessage() {
-    if (this.state === 'connected') {
-      return null;
-    }
-
-    if (this.state === 'disconnected') {
-      return {
-        session: this._session,
-        type: 'disconnect',
-        version: RSP_VERSION
-      };
-    }
-
-    const type = {
-      connecting: 'connect',
-      syncing: 'sync'
-    }[this.state];
-
-    const message = {
-      name: this._name,
-      participant: this._localParticipant.getState(),
-      peer_connections: this._peerConnectionManager.getStates(),
-      type,
-      version: RSP_VERSION
-    };
-
-    if (message.type === 'connect') {
-      message.ice_servers = this._iceServersStatus;
-
-      message.publisher = {
-        name: SDK_NAME,
-        sdk_version: SDK_VERSION,
-        user_agent: this._userAgent
-      };
-
-      if (this._bandwidthProfile) {
-        message.bandwidth_profile = createBandwidthProfilePayload(
-          this._bandwidthProfile);
-      }
-
-      message.media_signaling = createMediaSignalingPayload(
-        this._dominantSpeaker,
-        this._networkQuality,
-        this._trackPriority,
-        this._trackSwitchOff,
-        this._adaptiveSimulcast,
-        this._renderHints);
-
-      message.subscribe = createSubscribePayload(
-        this._automaticSubscription);
-
-      const sdpFormat = this._options.sdpFormat;
-      if (sdpFormat) {
-        message.format = sdpFormat;
+    // NOTE(mmalavalli): In order to hide the AccessToken from being available
+    // to the application code, we define those member functions that make use
+    // of it here.
+    Object.defineProperties(this, {
+      _createConnectOrSyncOrDisconnectMessage: {
+        value: () => createConnectOrSyncOrDisconnectMessage(this, accessToken)
```

Comment:
How about
```js
value: createConnectOrSyncOrDisconnectMessage.bind(this, accessToken)
```

or
```js
value: () => createConnectOrSyncOrDisconnectMessage.call(this, accessToken)
```

This way, you don't need to pass the transport object and don't have to do the big destructuring in line 564

Reply:
@charliesantos I'd like to avoid using `bind`, and be more explicit in defining the arguments required for a particular function. Otherwise, it is a bit confusing to read the function, where `this` is being used even though it is not a member of any class.